cli/tests/functional/src/test/java/io/helidon/tests/functional/CliMavenTest.java

```diff
@@ -0,0 +1,677 @@
+                        content = content.replaceAll("\"World\"", "\"Jhon\"");
+                        Files.write(path, content.getBytes(StandardCharsets.UTF_8));
+                    } catch (IOException ioException) {
+                        Assertions.fail("Unable to modified GreetService file");
+                    }
+                });
+
+        app.waitForApplication();
+
+        WebClient client = WebClient.builder()
+                .baseUri("http://localhost:" + port + "/greet")
+                .build();
+        client.get().request(String.class)
+                .thenAccept(s -> Assertions.assertTrue(s.contains("Jhon")))
+                .toCompletableFuture().get();
+
+        app.stop();
+    }
+
+    @Test
+    public void testVerbose() throws Exception {
+        int port = platform.getAvailablePorts().next();
+        List<String> cmdArgs = new LinkedList<>(List.of("--app-jvm-args", "-Dserver.port=" + port));
+
+        CliApplication generateApp = Builder.builder()
+                .mavenVersion(MAVEN_VERSION.V3_8_4)
+                .batch()
+                .build();
+        generateApp.runInitCommand();
+        generateApp.stop();
+
+        CliApplication app = Builder.builder()
+                .mavenVersion(MAVEN_VERSION.V3_8_4)
+                .port(port)
+                .verbose()
+                .build();
+        app.runDevCommand(cmdArgs);
+        app.waitForApplication();
+        app.stop();
+        CapturingApplicationConsole console = app.console();
+
+        if (console.getCapturedOutputLines()
+                .stream()
+                .noneMatch(line -> line.contains("Detecting the operating system and CPU architecture"))) {
+            Assertions.fail("Verbose mode does not print system information");
+        }
+    }
+
+    @Test
+    public void testDebug() {
+        CliApplication app = Builder.builder()
+                .mavenVersion(MAVEN_VERSION.V3_8_4)
+                .batch()
+                .debug()
+                .build();
+        app.runInitCommand();
+        app.stop();
+        CapturingApplicationConsole console = app.console();
+        console.getCapturedOutputLines().forEach(System.out::println);
+        if (console.getCapturedOutputLines()
+                .stream()
+                .noneMatch(line -> line.contains("Found maven executable"))) {
+            Assertions.fail("Debug mode does not print required information");
+        }
+    }
+
+    @Test
+    public void testCliMavenPluginJansiIssue() throws Exception {
+        int port = platform.getAvailablePorts().next();
+
+        CliApplication generateApp = Builder.builder()
+                .mavenVersion(MAVEN_VERSION.V3_8_1)
+                .batch()
+                .build();
+        generateApp.runInitCommand();
+        generateApp.stop();
+
+        setCustomServerPortInPom(generateApp.getGeneratedProjectLocation().resolve("pom.xml"), port);
```

Comment:
Can you explain why you have to edit the project pom ?

Reply:
The goal is to override the server port so it deploys the application to a random port.
I could not find a way to do it through the maven command, so I edit the pom to write plugin configuration with 
`appJvmArgs=-Dserver.port=0`. 
Is there an easier way to do it without editing the pom file ?

Reply:
`appJvmArgs` is exposed as a parameter in the `helidon dev` command:
```
  --app-jvm-args         JVM args used when starting the application
```

Reply:
But here I am not running `helidon dev`, I am running `mvn io.helidon.build-tools:helidon-cli-maven-plugin:3.0.0-SNAPSHOT:dev` and I cannot pass such parameter, or I did not find a way from the mvn command.

Reply:
```java
    /**
     * Application JVM arguments.
     */
    @Parameter(property = "dev.appJvmArgs")
    private String appJvmArgs;
```